File: baekjoon/Java/8320_직사각형을 만드는 방법.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int ans = N;
        for (int i = 2; i < N; i++){
            int tmp = N / i ;
            int n = tmp - (i-1);
            if(n < 1) break;
            ans += n;
        }
        System.out.println(ans);
    }
}

File: baekjoon/Python/8320_직사각형을 만드는 방법.py
# 방법1. 가로 세로 탐색
N = int(input())

ans = 0
for i in range(1, N + 1):
    for j in range(i, N + 1):
        if i*j <= N:
            ans += 1
print(ans)

# 방법2. 규칙
N = int(input())

ans = N
for i in range(2, N):
    n = N // i - (i-1)
    if n < 1:
        break
    ans += n
print(ans)
